config/runtimes/kserve-huggingfaceserver-multinode.yaml

```diff
@@ -0,0 +1,231 @@
+        
+      env:
+        - name: RAY_PORT
+          value: "6379"
+        - name: HOME
+          value: "/tmp"
+        - name: RAY_ADDRESS
+          value: 127.0.0.1:6379
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: TENSOR_PARALLEL_SIZE
+          value: "1"
+        - name: HF_HUB_CACHE
+          value: /tmp
+      resources:
+        limits:
+          cpu: "16"
+          memory: 48Gi
+          nvidia.com/gpu: "1"
+        requests:
+          cpu: "8"
+      volumeMounts:
+        - name: shm
+          mountPath: /dev/shm     
+      livenessProbe:
+        failureThreshold: 3
+        periodSeconds: 30
+        successThreshold: 1
+        timeoutSeconds: 5
+        exec:
+          command:
+            - bash
+            - -c
+            - |            
+              # Check GPU usage (If worker node restarted, it becomes NOT ready)
+              gpu_status=$(ray status | grep GPU)
+              used_gpu=$(echo "$gpu_status" | awk '{print $1}' | cut -d'/' -f1)
+              reserved_gpu=$(echo "$gpu_status" | awk '{print $1}' | cut -d'/' -f2)
+
+              # Determine health status based on GPU usage
+              if [[ "$used_gpu" != "$reserved_gpu" ]]; then
+                  echo "Liveness Probe: Unhealthy - Used: $used_gpu, Reserved: $reserved_gpu"
+                  exit 1
+              else
+                  exit 0                  
+              fi 
+      readinessProbe:
+        failureThreshold: 2
+        periodSeconds: 10
+        successThreshold: 1
+        timeoutSeconds: 5
+        exec:
+          command:
+            - bash
+            - -c
+            - |
+              # Check if the registered ray nodes count is the same as PIPELINE_PARALLEL_SIZE
+              if [[ $(ray status | grep -c node_) -eq ${PIPELINE_PARALLEL_SIZE} ]]; then
+                exit 0
+              else
+                exit 1
+              fi &
+              p1=$!
+
+              # Check if huggingface server health
+              curl --silent --max-time 5 http://localhost:8080 &
+              p2=$!
+
+              # Check GPU usage (If worker node restarted, it becomes NOT ready)
+              gpu_status=$(ray status | grep GPU)
+              used_gpu=$(echo "$gpu_status" | awk '{print $1}' | cut -d'/' -f1)
+              reserved_gpu=$(echo "$gpu_status" | awk '{print $1}' | cut -d'/' -f2)
+
+              # Determine health status based on GPU usage
+              if [[ "$used_gpu" != "$reserved_gpu" ]]; then
+                  echo "Readiness Probe: Unhealthy - Used: $used_gpu, Reserved: $reserved_gpu"
+                  exit 1
+              else
+                  exit 0                  
+              fi & # Run this check in the background
+              p3=$!
+
+              # Wait for all processes to complete
```

Comment:
I think it is better to fail fast, rather than run everything in the background.
E.g. it doesn't make sense to run `curl` is ray is not fully up.

Reply:
Alternatively we can implement this is the health handler in huggingserver runtime

Reply:
I update this part with manifest.

>Alternatively we can implement this in the health handler in huggingserver runtime

I am not sure it is a good idea to include this kind of health check login into servingruntime or not.

Reply:
I added a script for health check